src/dmd/statementsem.d

```diff
@@ -3225,7 +3225,12 @@ else
                     }
                     else if (rs.exp.op != TOK.error)
                     {
-                        rs.error("mismatched function return type inference of `%s` and `%s`", rs.exp.type.toChars(), tret.toChars());
+                        Loc prevretloc = (!fd.returns) ? fd.loc : (*fd.returns)[0].loc;
```

Comment:
Variable names should be camelCase according to [the D Style](https://dlang.org/dstyle.html#naming_general)

Reply:
Also I think it's more idiomatic to use `auto` for the storage class in this case.

Reply:
Fixed.